src/client/core/utils/dom.js

```diff
@@ -364,28 +367,78 @@ export function getElementDescription (el) {
     return res.join('');
 }
 
-export function getNextFocusableElement (element, reverse) {
-    const offset       = reverse ? -1 : 1;
+export function focusNextElement (element, reverse, skipRadioGroups) {
+    return new Promise(resolve => {
+        const nextElement = getNextFocusableElement(element, reverse, skipRadioGroups);
+
+        if (nextElement)
+            focusBlurSandbox.focus(nextElement, () => resolve(nextElement));
+        else
+            resolve();
+    });
+}
+
+function filterFocusableElements (elements, sourceElement, skipRadioGroups) {
+    if (!isRadioButtonElement(sourceElement))
+        return elements;
+
+    const isArrowNavigationAllowed                  = !skipRadioGroups && sourceElement.name !== '';
+    const isArrowNavigationBetweenNonamesAllowed    = !skipRadioGroups && !sourceElement.name && browserUtils.isChrome;
+    const isArrowNavigationBetweenNonamesDisallowed = !skipRadioGroups && !sourceElement.name && !browserUtils.isChrome;
```

Comment:
I find this logic unclear and a bit misleading. E.g. there are `...Allowed` and `...Disallowed` variables, but the relationship between them is not binary. Could you just use `else` cases in the conditions below?